cpmpy/expressions/core.py

```diff
@@ -454,6 +455,8 @@ def value(self):
 def _wsum_should(arg):
     """ Internal helper: should the arg be in a wsum instead of sum """
     # Undecided: -x + y, -x + -y?
+    if isinstance(arg, Operator) and arg.name == "-":
```

Comment:
I am not sure we should always convert a sum with a - into a wsum...
Ideally we do very few transformatiosn to the expression prior to calling 'flatten()', so that the debugprint of the model looks recognizable for the user.

So, are you sure we need this? Is the special case not handlable somewhere else?

Reply:
No we don't. I added the rewrite rules to the flattening.